### PR TITLE
Feat/exit code for no projects

### PR DESCRIPTION
**Pull request by lili2311**
- [x] Ready for review
- [x] Follows [CONTRIBUTING](https://github.com/snyk/snyk/blob/master/.github/CONTRIBUTING.md) rules
- [x] Reviewed by Snyk internal team

#### What does this PR do?
Add a new Exit Code when non supported manifests were detected so we could not conduct a test.
https://github.com/snyk/snyk/issues/1135

#### Where should the reviewer start?
https://github.com/snyk/snyk/compare/feat/exit-code-for-no-projects#diff-ebd6539fe4cc61b013d83b586f268b8aR93

#### How should this be manually tested?
`snyk test` run an empty folder for example

Screen Shot 2020-09-04 at 9 57 09

### CODE REVIEW

File: src/lib/snyk-test/index.js [hunk 1]
```diff
@@ -51,7 +51,9 @@ function executeTest(root, options) {
       return results;
     });
   } catch (error) {
-    return Promise.reject(chalk.red.bold(error));
+    return Promise.reject(
+      chalk.red.bold(error.message ? error.message : error),
```
Comment (lili2311) [inline src/lib/snyk-test/index.js:55]: fix to drop `Custom Error` from displayed errors.

**Comment by lili2311** [comment]
https://github.com/snyk/snyk/pull/1387

File: test/smoke/spec/snyk_test_spec.sh [hunk 2]
```diff
@@ -10,9 +10,23 @@ Describe "Snyk test command"
       snyk test
     }
 
+    It "throws error when file does not exist"
```
Comment (lili2311) [inline test/smoke/spec/snyk_test_spec.sh:13]: Test passing locally:
Screen Shot 2020-09-04 at 9 49 06

File: src/cli/index.ts [hunk 3]
```diff
@@ -43,6 +43,7 @@ const debug = Debug('snyk');
 const EXIT_CODES = {
   VULNS_FOUND: 1,
   ERROR: 2,
+  NO_SUPPORTED_MANIFESTS_FOUND: 3,
```
Comment (JackuB) [inline src/cli/index.ts:46]: Could you please also document this somewhere? Readme or help.txt

**Review by JackuB — APPROVED** [review]

File: src/cli/index.ts (hunk 3 above)
Comment (lili2311) [reply src/cli/index.ts:46]: Documented:

- https://support.snyk.io/hc/en-us/articles/360003812578
- https://snyk.zendesk.com/hc/en-us/articles/360006786558
- and in `snyk help`

**Pull request merged**